snapcraft/internal/sources/_git.py

```diff
@@ -90,3 +90,25 @@ def pull(self):
             self._pull_existing()
         else:
             self._clone_new()
+        self.assets = self._get_source_details()
+
+    def _get_source_details(self):
```

Comment:
It seems like this needs to be a public method the source instance for the part (pluginhandler) can call into at the appropriate time instead of at init.